From bff711a1ada55d84b5e6ca8b21d8a7f548c90bb9 Mon Sep 17 00:00:00 2001
From: Richard Zhu <Richard.Zhu@freescale.com>
Date: Fri, 9 Oct 2015 14:43:24 +0800
Subject: [PATCH 0809/5242] MLK-11681-2 ata: imx: enable sata on imx6qp

commit  69f7c9a531ee4678f07b4c8917dfcfd2df8810f6 from
https://source.codeaurora.org/external/imx/linux-imx.git

Since the version of the imx6qp's ahci controller
is same to the version of imx6q ahci controller.
So, this work-around should be applied to imx6qp
ahci too.

Signed-off-by: Richard Zhu <Richard.Zhu@freescale.com>
(cherry picked from commit 31a7eab5be9843263535306dbaa08592b66714e5)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/ata/ahci_imx.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/ata/ahci_imx.c b/drivers/ata/ahci_imx.c
index 3121a7d..393ab35 100644
--- a/drivers/ata/ahci_imx.c
+++ b/drivers/ata/ahci_imx.c
@@ -1201,7 +1201,7 @@ static int imx_ahci_probe(struct platform_device *pdev)
 	* which is present on mx6q, and not on mx53,
 	* we should use sg_tablesize = 1 for reliable operation
 	*/
-	if (imxpriv->type == AHCI_IMX6Q) {
+	if (imxpriv->type == AHCI_IMX6Q || imxpriv->type == AHCI_IMX6QP) {
 		dma_addr_t dma;
 
 		ahci_platform_sht.sg_tablesize = 1;
-- 
1.7.9.5

